### PR TITLE
Disable polling events & logs from Yorc server when the orchestrator is disabled

### PR DESCRIPTION
**Pull request by BertrandD**

### CODE REVIEW

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java [hunk 1]
```diff
@@ -47,11 +47,14 @@
...
-        return beanFactory.getBean(YorcPaaSProvider.class);
+        YorcPaaSProvider instance = beanFactory.getBean(YorcPaaSProvider.class);
+        instance.startLogsAndEvents();
+        return instance;
     }
 
     @Override
     public void destroy(YorcPaaSProvider instance) {
+        instance.stopLogsAndEvents();
         // nothing specific, the plugin will be garbaged collected when all references are lost.
```
Comment (loicalbertin) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java]: Now there is something specific :smile: we could remove this comment.

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java [hunk 2]
```diff
@@ -47,11 +47,14 @@
...
     @Override
     public YorcPaaSProvider newInstance() {
-        return beanFactory.getBean(YorcPaaSProvider.class);
+        YorcPaaSProvider instance = beanFactory.getBean(YorcPaaSProvider.class);
+        instance.startLogsAndEvents();
+        return instance;
     }
 
     @Override
     public void destroy(YorcPaaSProvider instance) {
```
Comment (loicalbertin) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java]: Is this method called when disabling an orchestrator or when destroying it?

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YorcPaaSProvider.java [hunk 3]
```diff
@@ -146,6 +148,21 @@ public YorcPaaSProvider() {
         taskManager = new TaskManager(3, 120, 3600);
     }
 
+    public void stopLogsAndEvents() {
+        eventListenerTask.valid = false;
```
Comment (loicalbertin) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YorcPaaSProvider.java]: This is not really "java-style" to set attributes directly.
I would use a setter or a `stop()` method that manages `valid` and make it private

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java (hunk 2 above)
Comment (BertrandD) [reply alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java]: On disabling: it is used in the `disable` method in the alien core code: https://github.com/alien4cloud/alien4cloud/blob/develop/alien4cloud-core/src/main/java/alien4cloud/orchestrators/services/OrchestratorStateService.java#L238

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YorcPaaSProvider.java (hunk 3 above)
Comment (BertrandD) [reply alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YorcPaaSProvider.java]: Ok done !

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java (hunk 1 above)
Comment (BertrandD) [reply alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/YstiaOrchestratorFactory.java]: :smile:

**Pull request merged**